### ISSUE TITLE
Could not get a validated dataspace when reading mesh on 8000 nodes

### ISSUE THREAD

**Issue by Thomas-Ulrich** [SeisSol/SeisSol#718]
**Describe the bug**
Code crashes when reading mesh when using ~8000 ranks.
```
Wed Oct 26 17:41:48, Info:  Running mini SeisSol to determine node weight
Wed Oct 26 17:41:49, Info:  Node weights: mean = 3.90598  std = 0.0304249  min = 3.69556  median = 3.9169  max = 3.93769
Wed Oct 26 17:41:49, Info:  Reading PUML mesh mesh_files/Ridgecrest_remeshed_f200_t200_nf_10Hz
Wed Oct 26 17:41:49, Info:  Found 178075455 cells
Wed Oct 26 17:41:52, Info:  Found 29984783 vertices
HDF5-DIAG: Error detected in HDF5 (1.10.7) MPI-process 7999:
  #000: /tmp/ulrich/spack-stage/spack-stage-hdf5-1.10.7-pxgykdc24h2fymekqdmg6cyctwkjcsxm/spack-src/src/H5Dio.c line 184 in H5Dread(): could not get a validated dataspace from file_space_id
    major: Invalid arguments to routine
    minor: Bad value
  #001: /tmp/ulrich/spack-stage/spack-stage-hdf5-1.10.7-pxgykdc24h2fymekqdmg6cyctwkjcsxm/spack-src/src/H5S.c line 257 in H5S_get_validated_dataspace(): selection + offset not within extent
    major: Dataspace
    minor: Out of range
Wed Oct 26 17:41:52, Error: An HDF5 error occurred in PUML (/scratch1/09160/ulrich/SeisSol/submodules/PUML/PUML.h: 270)
Backtrace:
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2(_ZN5utils6LoggerD1Ev+0x30c) [0x42536c]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x461c45]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x451d57]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x43feab]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x44920c]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x702a8b]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x62df4f]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x5cad87]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x5c951c]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x41da97]
/usr/lib64/libc.so.6(__libc_start_main+0xf5) [0x2ae8faa58555]
/scratch1/09160/sebwolf/launch_seissol/Ridgecrest_hyperparameters/SeisSol_Release_dskx_6_viscoelastic2() [0x41d8bf]
Abort(134) on node 7999 (rank 7999 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 134) - process 7999
[proxy:0:6097@c172-033.frontera.tacc.utexas.edu] wait_proxies_to_terminate (../../../../../src/pm/i_hydra/proxy/intel/i_proxy.c:124): downstream from host c172-053 exited with status 255
[proxy:0:5783@c168-092.frontera.tacc.utexas.edu] wait_proxies_to_terminate (../../../../../src/pm/i_hydra/proxy/intel/i_proxy.c:124): downstream from host c168-154 exited with status 255
```

**Expected behavior**

no crash.
**To Reproduce**
1. Which version do you use? Provide branch and commit id.
a71e59e7e3cbd1583f620d55a93f34176402bba6, fix-707-2-nuc
(adab0bbcbdc5b980b79fcbecb7b08a49d9478845 + double nucleation + fix 707)
2. Which build settings do you use? Which compiler version do you use? 
intel
```SeisSol_Release_dskx_6_viscoelastic2```
3. On which machine does your problem occur? If on a cluster: Which modules are loaded?
frontera.
```
  1) intel/19.1.1   2) impi/19.0.9   3) git/2.24.1   4) autotools/1.2   5) cmake/3.24.2   6) pmix/3.1.4   7) hwloc/1.11.12   8) xalt/2.10.34   9) TACC  10) python3/3.9.2  11) seissol-env-develop-intel-19.1.1.217-x52n3zf
```
4. Provide parameter/material files.
Ridgecrest setup

**Comment by sebwolf-de** [comment]
I would like to add that this did not happen on 7500 ranks, and was independent of the hdf5 version. I'll try to get a reference on SuperMUC with 96 ranks per node ;-)

**Comment by krenzland** [comment]
It did also happen on the last SuperMUC full machine run. So also independent of the machine.

**Comment by sebwolf-de** [comment]
I could reproduce this issue with 24 nodes, 48 ranks per node on SuperMUC-NG... Let's see what SuperMUC admins say.

**Comment by krenzland** [comment]
That is quite strange. I've used more than 24*48 mpi ranks on NG in the past?

**Comment by sebwolf-de** [comment]
Fun fact: Mesh with 27M cells works on 24*48 ranks. Mesh with 2.3M cells does not work.
I'm pretty convinced, the error comes from here: https://github.com/TUM-I5/PUML2/blob/master/PUML.h#L255-L270, because we are reading too small data chunks (or something like that, I'm not a hdf5 expert).

**Comment by krenzland** [comment]
Ha, nice. That should make it far easier!

**Comment by sebwolf-de** [comment]
Problem is here:
https://github.com/TUM-I5/PUML2/blob/master/PUML.h#L257
In the case, that Thomas has reported, we have `29984783` vertices. `m_originalSize[1]` is `3749`. So in total, we want to read `3749*8000 = 29992000` vertices, which is more than there are in the mesh :D

**Comment by sebwolf-de** [comment]
Fixed with #721
